sdk/storage/src/datalake/path_client.cpp

```diff
@@ -0,0 +1,440 @@
+
+    return PathClient(builder.to_string(), credential, options);
+  }
+
+  PathClient::PathClient(
+      const std::string& pathUri,
+      std::shared_ptr<SharedKeyCredential> credential,
+      const PathClientOptions& options)
+      : m_pathUri(pathUri)
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+    for (const auto& p : options.policies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    policies.emplace_back(std::make_unique<CommonHeadersRequestPolicy>());
+    policies.emplace_back(std::make_unique<SharedKeyPolicy>(credential));
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
+    m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
+  }
+
+  PathClient::PathClient(
+      const std::string& pathUri,
+      std::shared_ptr<TokenCredential> credential,
+      const PathClientOptions& options)
+      : m_pathUri(pathUri)
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+    for (const auto& p : options.policies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    policies.emplace_back(std::make_unique<CommonHeadersRequestPolicy>());
+    // not implemented yet
+    unused(credential);
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
+    m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
+  }
+
+  PathClient::PathClient(const std::string& pathUri, const PathClientOptions& options)
+      : m_pathUri(pathUri)
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+    for (const auto& p : options.policies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    policies.emplace_back(std::make_unique<CommonHeadersRequestPolicy>());
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
+    m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
+  }
+
+  PathAppendDataResponse PathClient::AppendData(
+      std::vector<uint8_t> data,
+      int64_t position,
+      const PathAppendDataOptions& options) const
+  {
+    DataLakeRestClient::Path::AppendDataOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.Body = std::move(data);
+    protocolLayerOptions.Position = position;
+    protocolLayerOptions.ContentLength = data.size();
+    protocolLayerOptions.TransactionalContentMD5 = options.ContentMD5;
+    protocolLayerOptions.LeaseIdOptional = options.LeaseId;
+    protocolLayerOptions.Timeout = options.Timeout;
```

Comment:
This seems really well-suited for IIFE.

Reply:
Can you please elaborate on the necessity of using IIFE here?